### ISSUE TITLE
Individual repository pages will not load if there was a release before the first Github star

### ISSUE THREAD

**Issue by Lance-Drane** [LLNL/llnl.github.io#531]
When navigating to an individual repository page, the "Number of Stars Over Time" chart will sometimes fail to load, and the browser will eventually warn that a script is slowing it down (due to an infinite loop occurring in the code).

Example page where this issue happens: https://software.llnl.gov/repo/#/LLNL/Aluminum

This is because the first star date is `2018-10-01`, but the first release timestamp is `2018-09-14T22:15:01Z` .

The specific block of code where the infinite loop occurs is here, in `js/explore/line_repoStarHistory.js`:
```
// Process release data to prevent clumps
for (var i = 0; i < releaseData[repoNameWOwner]['dates'].length; i++) {
    var date = parseTime(releaseData[repoNameWOwner]['dates'][i].slice(0, releaseData[repoNameWOwner]['dates'][i].indexOf('T')));
    current = x(date);
    if (start < 0) {
        start = current;
        past = i;
        i--;
    } else if (current - start < tolerance) {
        past = i;
    } else {
        indexArray.push(past);
        start = -1;
        current = -1;
        past = -1;
        i--;
    }
}
```

On the first loop iteration, the code will get stuck due to `i` continually being reset to 0. The `if` block will always execute successfully.

I would be happy to submit a PR to resolve this issue.

**Comment by IanLee1521** [comment]
Thank you for submitting this @Lance-Drane !

**Comment by LRWeber** [comment]
Resolved via #532
